front/components/agent_builder/AgentBuilder.tsx

```diff
@@ -230,10 +230,14 @@ export default function AgentBuilder({
         type: "success",
       });
 
+      // Reset form dirty state after successful save
+      form.reset(form.getValues(), {
+        keepValues: true,
+      });
+
       if (!agentConfiguration && createdAgent.sId) {
         const newUrl = `/w/${owner.sId}/builder/agents/${createdAgent.sId}`;
-        // willingly using window to not trigger next navigation events
-        window.history.replaceState(null, "", newUrl);
+        await router.push(newUrl);
```

Comment:
You could use the following if you'd like the same behaviour as before:
```suggestion
        await router.replace(newUrl, undefined, { shallow: true });
```